lib/wifi_user/use_case/email_signup.rb

```diff
@@ -2,16 +2,17 @@
 require "notifications/client"
 
 class WifiUser::UseCase::EmailSignup
-  def initialize(user_model:, allowlist_checker:, logger: Logger.new($stdout))
+  def initialize(user_model:, allowlist_checker:, check_user_is_sponsee:, logger: Logger.new($stdout))
     @user_model = user_model
     @allowlist_checker = allowlist_checker
+    @check_user_is_sponsee = check_user_is_sponsee
     @logger = logger
   end
 
   def execute(contact:)
     email_address = Mail::Address.new(contact).address
 
-    if allowlist_checker.execute(email_address)[:success]
+    if allowlist_checker.execute(email_address)[:success] || check_user_is_sponsee.execute(email_address)
```

Comment:
The main problem here is that this will render the wrong template - it sends the standard user signup message and it probably should send the sponsor signup one